File: LeetCode_3228/Main.py
class Solution:
    def maxOperations(self, s: str) -> int:
        ones = operation = 0
        n = len(s)

        for i in range(n - 1):
            if s[i] == '0' and s[i + 1] == '1':
                operation += ones
            
            if s[i] == '1':
                ones += 1
        if s[n - 1] == '0':
            operation += ones
        return operation

File: LeetCode_3228/Main.cpp
#include<bits/stdc++.h>
using namespace std;
#define edl '\n'

class Solution {
public:
    int maxOperations(string s) {
        int ones = 0, operation = 0;

        for (int i = 0; i + 1 < s.size(); i++) {
            if (s[i] == '0' && s[i + 1] == '1') {
                operation += ones;
            }
            ones += s[i] == '1';
        }

        if (s.back() == '0') {
            operation += ones;
        }
        return operation;
    }
};